plugin/storage/es/spanstore/reader.go

```diff
@@ -201,16 +201,13 @@ func addRemoteReadClusters(fn timeRangeIndexFn, remoteReadClusters []string) tim
 	}
 }
 
-func getSourceFn(archive bool, maxDocCount int) sourceFn {
+func getSourceFn(maxDocCount int) sourceFn {
 	return func(query elastic.Query, nextTime uint64) *elastic.SearchSource {
-		s := elastic.NewSearchSource().
+		return elastic.NewSearchSource().
 			Query(query).
-			Size(maxDocCount)
-		if !archive {
-			s.Sort("startTime", true).
-				SearchAfter(nextTime)
-		}
-		return s
+			Size(maxDocCount).
+			Sort("startTime", true).
+			SearchAfter()
```

Comment:
doesn't seem equivalent if you're excluding `nextTime` for primary storage.

Reply:
@yurishkuro yeah its not - i was just trying it to see how the CI reacts. I'm a little confused - why do v1 and v2 tests have different behaviour? Before I removed this parameter, v1 tests were failing and v2 was passing and now its the opposite.

Reply:
I don't have a general answer. The mechanics of v1/v2 test are very different.